lib/widgets/widgets/zabbix/zabbix_widget.dart

```diff
@@ -0,0 +1,381 @@
+  void dispose() {
+    this._controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  void didUpdateWidget(ZabbixWidget oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (previousValue != int.parse(getLastValue)) {
+      double begin = this.valueTween?.evaluate(this._controller) ?? calculatePercentageValue(previousValue) ?? 0;
+      this.valueTween = Tween<double>(
+        begin: begin,
+        end: calculatePercentageValue(int.parse(getLastValue)) / 100,
+      );
+      this._controller
+        ..value = 0
+        ..forward();
+    }
+  }
+
+  List get getHistory {
+    Map historyMap = widget.widget.content['history'];
+    return historyMap.entries
+        .map((e) => ZabbixHistoryItem(
+              timestamp: e.key,
+              value: e.value,
+            ))
+        .toList();
+  }
+
+  bool get displayUpArrow {
+    List historyData = getHistory;
+    int current = int.parse(historyData[historyData.length - 1].value);
+    int prev = int.parse(historyData[historyData.length - 2].value);
+    return current > prev;
+  }
+
+  String get getTitle {
+    return ZabbixMetrics[widget.widget.selectedZabbixMetric];
+  }
+
+  int get getMaxValue {
+    return widget.widget.maxValue;
+  }
+
+  int get rangeLowerBound {
+    return widget.widget.range[0];
+  }
+
+  int get rangeUpperBound {
+    return widget.widget.range[1];
+  }
+
+  String get getProgressType {
+    return checkMetricHasProgress() && (!checkMetricHasMaxValue() || widget.widget.maxValue > 0)
+        ? 'progress'
+        : 'maxValue';
+  }
+
+  bool get isSystemUptime {
+    return widget.widget.content['name'] == "System uptime";
+  }
+
+  String get getLastValue {
+    return widget.widget.content['lastvalue'];
+  }
+
+  bool checkMetricHasMaxValue() {
+    return ZabbixMetricsWithMaxValue.contains(widget.widget.selectedZabbixMetric);
+  }
+
+  bool checkMetricHasProgress() {
+    return ZabbixMetricsWithProgress.contains(widget.widget.selectedZabbixMetric);
+  }
+
+  int convertToBytes(int value) {
+    return ((100 * value) / (widget.widget.maxValue * pow(10, 9))).round();
+  }
+
+  int calculatePercentageValue(int value) {
+    if (!checkMetricHasMaxValue()) return value;
+    return convertToBytes(value);
+  }
+
+  int convertToGigabytes() {
+    if (getLastValue == null) return 0;
+    return (int.parse(getLastValue) / pow(10, 9)).round();
+  }
+
+  String secondsToTime(int value) {
+    var days = (value / 3600 / 24).floor().toString();
+    var hours = ((value / 3600) % 24).floor().toString();
+    var minutes = ((value % 3600) / 60).floor().toString();
+    return '${days}d:${hours}h:${minutes}m';
+  }
+
+  String get getNoProgressContent {
+    return widget.widget.selectedZabbixMetric == 'system.uptime'
+        ? secondsToTime(int.parse(getLastValue))
+        : int.parse(getLastValue).toString();
+  }
+
+  String get xAxisUnit {
+    if (checkMetricHasMaxValue()) {
+      return '[GB]';
+    } else if (!checkMetricHasProgress()) {
```

Comment:
🥇 [NOW] Change the condition and method names so it is more meaningful.
```
if (isInBytes()) {
      return '[GB]';
    } else if (isPercentage()) {
      return '[%]';
    } else {
    return 'No.';
    }
```

Reply:
done